system/lib/wasmfs/syscalls.cpp

```diff
@@ -513,4 +515,67 @@ __wasi_errno_t __wasi_fd_fdstat_get(__wasi_fd_t fd, __wasi_fdstat_t* stat) {
   }
   return __WASI_ERRNO_SUCCESS;
 }
+
+long __syscall_getdents64(long fd, long dirp, long count) {
+  auto openFile = wasmFS.getLockedFileTable()[fd];
+
+  if (!openFile) {
+    return -EBADF;
+  }
+  dirent* result = (dirent*)dirp;
+
+  // Check if the result buffer is too small.
+  if (count / sizeof(dirent) == 0) {
+    return -EINVAL;
+  }
+
+  auto file = openFile.locked().getFile();
+
+  auto directory = file->dynCast<Directory>();
+  // Check if the file is a directory.
```

Comment:
In general, I think the previous comment is good - it's not obvious when reading the `count / sizeoff` stuff what the meaning is.

But this type of comment seems redundant: the code literally does `if (!directory)`, so I would remove it. Just phrasing the code in human language doesn't add much, I think, unless it also adds context or information.